Automattic/wp-calypso
# Plans: Use PlansSkipButton for WP.com Plans

Previously (as of #22384), we were using the entire `plans` array as passed to the `PlanFeatures` component, and simply filtering to yield only the free plan. We would then pass a number of attributes obtained that way to the `PlanActions` component, which would render the 'Start with Free' button. Out of all the props passed to that component, the only relevant one seems to be `onUpgradeClick` -- the click handler (which is somewhat of a misnomer for free plans).

Still inside the `PlanFeatures` component, we would then move on and create a React portal to place the button at the bottom of its parent `PlansFeaturesMain` component (this is what the `PlanFeaturesBottom` component was for).

This PR uses the `PlansSkipButton` component instead, rendering it directly into `PlansFeaturesMain`. However, since we're no longer going through the `PlanFeatures` component, we lose the [additional wrapping](https://github.com/Automattic/wp-calypso/blob/34755834a2d4b43b58b0f6cf572bde8f886449a6/client/my-sites/plan-features/index.jsx#L819-L831) done to the `onUpgradeClick` prop in its `connect()` statement -- which essentially just binds the selected plan as a cart item to it. However, we can easily compensate for that by passing `null` to `onUpgradeClick`, which is the value that represents the free plan.

### Design Question
Is it okay to keep the arrow on the button? We're using it during Jetpack signup, so for consistency's sake, it might be nice to also have it for WP.com signup. (Otherwise, I'll have to add a prop to hide it).

#### Before

![image](https://user-images.githubusercontent.com/96308/39055724-36fab062-44b5-11e8-8a4b-c963d69b7350.png)

#### After

![image](https://user-images.githubusercontent.com/96308/39002427-d25a51d8-43f8-11e8-86aa-56b28f279af9.png)

### Testing Instructions
Verify that the `PlansFeaturesMain` component (i.e. the `plans-features-main` file) is only imported and used by the following files:

1. `jetpack-connect/plans-grid`: This is in turn used by
    1. `jetpack-connect/plans-landing` and
    1. `jetpack-connect/plans`
1. `my-sites/plans/main`
1. `signup/steps/plans`

These map to the following routes/testing scenarios:

In the following _logged-out_ JP Connect/Signup cases, verify that during the plans selection step, the 'Start with Free' button is present at the bottom of the page, and that it works:

1. WP.com Signup -- Create a new WP.com site (starting at calypso.localhost:3000/start)
1. Jetpack Connect: Create a new self-hosted WordPress with Jetpack installed (e.g. using jurassic.ninja), and connect to WP.com.

Logged-in:

1. Verify the same is true for _logged-in_ Jetpack Connect.
1. Note that in the _logged-in WP.com case_, we're displaying a 'Free Plan' column instead of the 'Start with Free' button. (Verify by using the site switcher to 'Add a new WordPress'.)

Finally, verify that the Plans (Overview) at `calypso.localhost:3000/plans/<site>` has _neither_ a 'Free Plan' column, nor a 'Start with Free' button. (Test for both a WP.com and a Jetpack site.)
